backend/frontend/src/utils/webapi.js

```diff
@@ -114,8 +115,12 @@ var posts = new Map();
 var cached = null;
 function getPiazzaPosts(courseId: number){
   let endpoint = mainUrl + `/${courseId}/course/getpiazzaposts/`;
-  http.get(endpoint, function(res){
+  let ignoredata = false;
+  var req = http.get(endpoint, function(res){
     res.on('data', function(buf){
+      if(ignoredata){
```

Comment:
What happens if someone leaves a piazza ID blank for a course? Does that mean you'll never do the API call to begin with?